torch/nn/utils/rnn.py

```diff
@@ -130,19 +147,20 @@ def pad_packed_sequence(sequence, batch_first=False, padding_value=0):
         padding_value (float, optional): values for padded elements.
 
     Returns:
-        Tuple of Variable containing the padded sequence, and a list of lengths
-        of each sequence in the batch.
+        Tuple of Variable containing the padded sequence, and Variable
+        containing the list of lengths of each sequence in the batch.
+
     """
```

Comment:
Here is where the BC code if a user feeds us a `list` as `batch_sizes` should live; we should generate a Variable LongTensor on the fly in that case.

Reply:
I had some backwards-compat code that worked a little differently earlier, and then was asked to take it out (I previously had two attributes, batch_size_as_list and batch_size_as_variable). I'll add back the version you're asking for here